.github/workflows/e2e.yaml

```
@@ -49,7 +49,8 @@ jobs:
           CLUSTERS_ARGS: ${{ matrix.globalnet }}
           DEPLOY_ARGS: ${{ matrix.globalnet }} --deploytool ${{ matrix.deploytool }}
         run: |
-          [[ -z "${{ matrix.cable_driver }}" ]] || DEPLOY_ARGS+=" --deploytool_submariner_args '--cable-driver ${{ matrix.cable_driver }}'"
+          [[ -z "${{ matrix.cable_driver }}" ]] || [[ ${{ matrix.deploytool }} == "helm" ]] || DEPLOY_ARGS+=" --deploytool_submariner_args '--cable-driver ${{ matrix.cable_driver }}'"
+          [[ -z "${{ matrix.cable_driver }}" ]] || [[ ${{ matrix.deploytool }} == "operator" ]] || DEPLOY_ARGS+=" --deploytool_submariner_args '--set cable-driver=${{ matrix.cable_driver }}'"
```

The conditions are confusing the way the are written, I know about boolean logic but it's hard for humans :D, I propose something like this
```suggestion
         if [[ -z "${{ matrix.cable_driver }}" ]]; then
              [[ ${{ matrix.deploytool }} == "operator" ]] && DEPLOY_ARGS+=" --deploytool_submariner_args '--cable-driver ${{ matrix.cable_driver }}'"
              [[ ${{ matrix.deploytool }} == "helm" ]] && DEPLOY_ARGS+=" --deploytool_submariner_args '--set cable-driver=${{ matrix.cable_driver }}'"
         fi 
```

---

Even better with a case/switch in bash?

---

as I remember I tried it, and it did work, let's see now.